amp-pwa-reader/src/server/server.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const express = require('express');
+const request = require('request');
+const fs = require('fs');
+const path = require('path');
+const cors = require('cors');
+const memCache = require('memory-cache');
+const pubBackend = require('./Backend.js');
+
+const app = express();
+const pub = new pubBackend();
+
+// how long (in seconds) to cache requests for main feed and for any article
+const cacheDurations = {feed: 600, article: 3600};
+const feedURL = 'https://query.yahooapis.com/v1/public/yql';
+
+// Whitelist domains for CORS. As more CDNs cache this site, we'll need to add them here.
+const corsOptions = {'origin': [
+  /amp\.cards$/,
+  /^localhost:/,
+  /amp-cards\.cdn\.ampproject\.org$/
+]};
+
+const staticFilesMiddleware = express.static('dist');
+app.use(staticFilesMiddleware);
+
+// Proxy the feed request so that we can cache it.
+// YQL expects a query in the "q" parameter.
+app.get('/feed', cache(cacheDurations.feed), function(req, res, next) {
+  const options = {
+    url: feedURL + '?format=json&q=' + req.query.q
+  };
+
+  request(options, (error, response, body) => {
+    if (!error) {
+      res.send(body);
+    } else {
+      res.json({error: 'An error occurred in /feed route'});
+    }
+  });
+});
+
+// This is used when the PWA requests a new article.
+// We proxy this request so that we can inject CORS headers and we can cache.
+app.get('/article', cache(cacheDurations.article), cors(corsOptions), function(req, res, next) {
```

Comment:
what's the diff between this route and the one below? Both say article.

Reply:
Which one below says article?  Can you clarify?

Reply:
In line 75, the comment says that this is also about articles...that confused me.